src/main/java/com/google/api/codegen/util/SymbolTable.java

```diff
@@ -70,17 +85,27 @@ public String getNewSymbol(String desiredName) {
     return desiredName + suffix;
   }
 
-  /**
-   * Returns the next numeric suffix that makes desiredName unique.
-   *
-   * <p>Stores the joined desiredName/suffix in an internal map. For example, if "foo" is passed, ""
-   * is returned. If "foo" is passed again, "2" is returned, and then "3" and so on.
-   */
   private String getAndSaveSuffix(String desiredName) {
     if (!symbolTable.contains(desiredName)) {
       symbolTable.add(desiredName);
       return "";
     }
+    if (disambiguationStrategy == DisambiguationStrategy.NUMERIC) {
```

Comment:
Consider a switch statement for clarity, especially if we grow the number of strategies.